docs/chymyst07.md

```diff
@@ -728,4 +728,104 @@ In the chemical machine, input molecules to each reaction must be defined static
+tobacco. We may give names to the smokers/workers as Keith, Slash, and Jimi. 
+
+The pusher provides at random time intervals one unit each of two required ingredients (for example matches and paper, or paper and tobacco). The pusher is not allowed to coordinate with the smokers to use knowledge of which smoker 
+needs which ingredients, he just supplies two ingredients at a time on a conveyor belt. We assume that the pusher has an infinite stream of the three 
+ingredients available to him. The real life example is that of a core operating systems service having to schedule and provide limited distinct resources to 
+other services where coordination of scarce resources is required.
+ 
+Each smoker selects two ingredients, rolls up a cigarette using the third ingredient that complements the list, lits it up and smokes it. It is necessary
+ for the smoker to finish his cigarette before the pusher supplies the next two ingredients (a timer can simulate the smoking activity).
+
+
+We model the processes as follows:
+
+| Supplier   | | Smoker 1   | |  Smoker 2    | |  Smoker 3 |
+| --- | --- | --- | --- | --- | --- |  --- |
+| select 2 random ingredients | | pick tobacco and paper | | pick tobacco and matches | | pick matches and paper |
+
+
+Let us now figure out the chemistry that will solve this problem. We can think of the problem as a concurrent producer consumer queue with three competing 
+consumers and the supplier simply produces random pairs of ingredients into the queue. For simplicity, we assume the queue has capacity 1, which is an 
+assumption in the statement of the problem.
+
+It also helps to draw parallels with the Dining Philosophers problem realizing that we can think of three philosophers as consumers and the constraint on 
+matching up the three ingredients differently for the three smokers is a bit analog to the 
+ 
+It is important to think of a suitable data model to capture the state of the world for the problem, so we need to know when to stop and count how many 
+cycles we go through, if we want to stop the computation. It may be useful to keep track of how many ingredients have been shipped or consumed but this does 
+not look to be important for a minimal solution (it is included for readers interested in inventory cost tracking and to represent that the state is carried 
+across the reactions and is not stored as mutable data anywhere).
+
+It is important to represent the ingredients as individual molecules and not as pairs of molecules to properly represent the 3-way dependence of the smokers 
+on the pairs of ingredients. We will represent the ingredients dynamically by representing the producer consumer queue, so we call the molecules shipments.
+
+One must ensure that we satisfy the chemistry laws, notably avoid unavoidable nondeterminism, which can occur if two distinct reactions in the same site 
```

Comment:
You cannot fail to satisfy the chemistry laws.  Also, you can't "avoid the unavoidable." :) 
What you wanted to say here is that your first solution had 3 reactions consuming the same molecule, but `Chymyst` doesn't like that and calls that "unavoidable nondeterminism". But the readers don't know that you had a first solution to the problem that didn't work. So this entire paragraph is going to be cryptic to the readers. I suggest removing it altogether. Instead, maybe just explain what each reaction does.